netstate/routes/netstate-routes.go

```diff
@@ -39,13 +40,23 @@ func NewNetStateRoutes(logger *zap.Logger, db *boltdb.Client) *NetStateRoutes {
 func (n *NetStateRoutes) Put(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	n.logger.Debug("entering netstate http put")
 
+	// check xapiheader sent from client to the one set
+	xAPIHeader := r.Header.Get("X-Api-Key")
```

Comment:
I'm wondering if we can do something here to eliminate the duplication.
maybe
```
func (n *NetStateRoutes) Validate(r *http.Request) error {
    xAPIHeader := r.Header.Get("X-Api-Key")
    if !auth.ValidateAPIKey(xAPIHeader) {
	http.Error(w, ErrInvalidCredentials, http.StatusUnauthorized)
	n.logger.Error(ErrInvalidCredentials)
	}
    return ErrInvalidCredentials
}
```

Reply:
I can move this outside as a function, and call the function from each route.